src/librustc_errors/json.rs

```diff
@@ -417,8 +418,14 @@ impl DiagnosticCode {
                 DiagnosticId::Error(s) => s,
                 DiagnosticId::Lint(s) => s,
             };
-            let explanation =
-                je.registry.as_ref().and_then(|registry| registry.find_description(&s));
+            let reg_result =
+                je.registry.as_ref().and_then(|registry| Some(registry.find_description(&s)));
+            let reg_result = reg_result.unwrap();
+
+            let explanation = match reg_result {
+                HasLongDescription(s) => Some(s),
+                _ => None,
```

Comment:
Do we need to rework this struct to also distinguish between non-existent long descriptions and invalid error codes? Currently it is pretty much treating it just as it was before, where a error code will be `Some(s)` and a code not existing / no long description will be `None`

Reply:
Ah, we should actually be able to panic or so if there's an invalid error code getting passed in here, that should never happen (if it does in practice we can fix it in this PR).

Reply:
I don't have a great idea of what this file is doing, what would be a good panic message?

Reply:
Doesn't matter too much. Just `panic!("unexpected invalid error code {}", code);` would be fine.